Automattic/wp-calypso
# Tests: Move Signup tests to single test runner

This pull request is part of https://github.com/Automattic/wp-calypso/issues/3942. It updates the unit test suites of the signup components to use the `client` test runner:

##### `client/lib/signup`

```
$ make test

  SignupDependencyStore
    ✓ should return an empty object at first
    ✓ should not store dependencies if none are included in an action
    ✓ should store dependencies if they are provided in either signup action

  SignupFlowController
    controlling a simple flow
      ✓ should run the onComplete callback with the flow destination when the flow is completed
    controlling a flow w/ an asynchronous step
      ✓ should call apiRequestFunction on steps with that property
      ✓ should not call apiRequestFunction multiple times
    controlling a flow w/ dependencies
      ✓ should call the apiRequestFunction callback with its dependencies
      ✓ should throw an error when the flow is completed without all dependencies provided
    controlling a flow w/ a delayed step
      ✓ should submit steps with the delayApiRequestUntilComplete once the flow is complete
      ✓ should not submit delayed steps if some steps are in-progress

  SignupProgressStore
    ✓ should return an empty at first
    ✓ should store a new step
    ✓ should not store the same step twice
    ✓ should not be possible to mutate
    ✓ should store multiple steps in order
    ✓ should mark submitted steps without an API request method as completed
    ✓ should mark submitted steps with an API request method as pending
    ✓ should mark only new saved steps as in-progress
    ✓ should set the status of a signup step
    timestamps
      ✓ should be updated at each step

  20 passing (223ms)
```

##### `client/signup`

```
$ make test

  flows.js
    ✓ should return the full flow when the user is not logged in
    ✓ should remove the user step from the flow when the user is not logged in

  utils.getLocale
    ✓ should find the locale anywhere in the params
    ✓ should return undefined if no locale is present in the params

  utils.getStepName
    ✓ should find the step name in either the stepName or flowName fragment
    ✓ should return undefined if no step name is found

  utils.getFlowName
    ✓ should find the flow name in the flowName fragment if present
    ✓ should return the default flow if the flow is missing
    ✓ should return the result of filterFlowName if it is a function and the flow is missing
    ✓ should return the result of filterFlowName if it is a function and the flow is not valid
    ✓ should return the result of filterFlowName if it is a function and the requested flow is present
    ✓ should return the passed flow if the result of filterFlowName is not valid
    ✓ should call filterFlowName with the default flow if it is a function and the flow is not valid
    ✓ should call filterFlowName with the requested flow if it is a function and the flow is valid

  utils.getValidPath
    ✓ should redirect to the default if no flow is present
    ✓ should redirect to the current flow default if no step is present
    ✓ should redirect to the default flow if the flow is the default
    ✓ should redirect invalid steps to the default flow if no flow is present
    ✓ should preserve a valid locale to the default flow if one is specified
    ✓ should redirect invalid steps to the current flow default
    ✓ should preserve a valid locale if one is specified
    ✓ should handle arbitrary step section names
    ✓ should handle arbitrary step section names in the default flow

  utils.getValueFromProgressStore
    ✓ should return the value of the field if it exists
    ✓ should return null if the field is not present

  utils.mergeFormWithValue
    ✓ should return the form with the field added if the field doesn't have a value
    ✓ should return the form unchanged if there is already a value in the form


  27 passing (263ms)
```

##### `client/signup/config`

```
$ make test

  signup/config
    ✓ should not have overlapping step/flow names

  1 passing (335ms)
```

#### Testing instructions
 
1. Run `git checkout update/signup-tests`
2. Execute `make test` and check that all tests for each module pass
